.github/workflows/update_dev_version.yml

```diff
@@ -14,31 +16,50 @@ jobs:
           ref: master
           path: main
 
+      - name: Checkout PennyLane-Lightning PR
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.AUTO_UPDATE_VERSION_RINGO_TOKEN }}
+          path: pr
+
+      - name: Checkout PR branch
+        env:
+          GITHUB_TOKEN: ${{ secrets.AUTO_UPDATE_VERSION_RINGO_TOKEN }}
+        run: cd pr && gh pr checkout ${{ github.event.pull_request.number }}
+
       - uses: actions/setup-python@v5
         name: Install Python
         with:
           python-version: '3.9'
 
-      - name: Checkout PennyLane-Lightning PR
-        uses: actions/checkout@v4
-        with:
-          token: ${{ secrets.GITHUB_TOKEN }}
-          ref: ${{ github.head_ref }}
-          path: pr
+      - name: Install Python Dependencies
+        run: pip install semver
+
+      - name: Capture Current Version in Pull Request
+        id: old_version
+        run: echo "version=$(grep -o '"[^"]\+[0-9]\+"' pr/pennylane_lightning/core/_version.py | sed 's/"//g')" >> $GITHUB_OUTPUT
```

Comment:
Should we validate that a number was captured here?

Reply:
Potentially, yes? Though I didn't think it was necessary as if there wasn't a valid semver in the _version.py file, then the subsequent step would error out and this value wouldn't matter anyways.